### PR TITLE
add extensions to DllReferencePlugin (#2892)

### PR DESCRIPTION
**Pull request by moo3**
**Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
[ Updated examples which has the README for each example updated ]

**What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
DllReferencePlugin / DllPlugin does not respect file extensions found in manifest.json
https://github.com/webpack/webpack/issues/2892

**What is the new behavior?**
Extension information is set to be exported as part of the manifest. The extensions information found in manifest can now be consumed (or overridden) by DllReferencePlugin.

**Does this PR introduce a breaking change?**
- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the following... 

* Impact:
* Migration path for existing applications: 
* Github Issue(s) this is regarding:

**Other information**:
The extensions packaged as part of manifest is honored only when DllReferencePlugin using _scope_ to resolve in the consuming end. If the DllReferencePlugin uses _context_ to resolve the resources in manifest then extensions information found in manifest is not used. Hence its important to note that when DllReferencePlugin is set to use a context, then resolve.extensions must be set for the plugin to resolve any resources of non-default type.

### CODE REVIEW

**Comment by sokra** [comment]
Could you remove the example README.md update from the PR?

File: lib/LibManifestPlugin.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ LibManifestPlugin.prototype.apply = function(compiler) {
 			var manifest = {
 				name: name,
 				type: this.options.type,
+				extensions: compiler.options.resolve.extensions,
```
Comment (sokra) [inline lib/LibManifestPlugin.js]: Plugins are not allow to access this property. Just remove `extensions` from the manifest.

File: lib/DllReferencePlugin.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ DllReferencePlugin.prototype.apply = function(compiler) {
 		type: this.options.type,
 		scope: this.options.scope,
 		context: this.options.context,
-		content: this.options.content || this.options.manifest.content
+		content: this.options.content || this.options.manifest.content,
+		extensions: this.options.extensions || this.options.manifest.extensions
```
Comment (sokra) [inline lib/DllReferencePlugin.js]: extensions should not be in the manifest

File: test/configCases/dll-plugin/1-use-dll/index.js [hunk 3]
```diff
@@ -6,6 +6,10 @@ it("should load a module from dll", function() {
 	require("dll/a").should.be.eql("a");
 });
 
+it("should load a module of non-default type without extension from dll", function() {
+	require("dll/f").should.be.eql("f");
+});
+
```
Comment (sokra) [inline test/configCases/dll-plugin/1-use-dll/index.js]: Pass extensions to the DllReferencePlugin in 1-use-dll

**Comment by sokra** [comment]
Hint: reference issues with `#2892` or `webpack/webpack#2892` instead of `webpack/2892`.

**Comment by moo3** [comment]
Updated with changes as requested. For some reason the _appveyor_ error is persistent. Let me know if I should do anything else.

**Comment by sokra** [comment]
merged

**Pull request closed without merging**